### PR TITLE
Masking in teca bard

### PR DESCRIPTION
**Pull request by taobrienlbl**
Adds the ability to mask input variables when running TECA BARD: fields are multiplied by `ar_probability`.  The use case here is calculating quantities like AR precipitation.

### CODE REVIEW

File: alg/teca_apply_binary_mask.cxx [hunk 1]
```diff
@@ -30,6 +54,94 @@ teca_apply_binary_mask::teca_apply_binary_mask() : mask_variable("")
...
+        }
+        else
+        {
+            // copy the attributes from the input. this will capture the
+            // data type, size, units, etc.
+            teca_array_attributes output_atts(input_atts);
+
+            // update description.
+            output_atts.description = 
+                std::string("masked/weighted by `" + this->mask_variable + "`");
```
Comment (burlen) [inline alg/teca_apply_binary_mask.cxx:135]: we should also update the long_name

File: alg/teca_apply_binary_mask.cxx [hunk 2]
```diff
@@ -30,6 +54,94 @@ teca_apply_binary_mask::teca_apply_binary_mask() : mask_variable("")
...
+{
+#ifdef TECA_DEBUG
+    cerr << teca_parallel_id()
+        << "teca_apply_binary_mask::get_output_metadata" << endl;
+#endif
+    (void)port;
+
+    if (this->input_variables.empty())
+    {
+        TECA_WARNING("The list of input variables was not set")
```
Comment (burlen) [inline alg/teca_apply_binary_mask.cxx:103]: might be good to warn only on rank 0 for large parallel runs

**Review by burlen — APPROVED** [review]
I saw a few things, most of them minor. the main points were that by processing teca_mesh(instead of teca_cartesian_mesh) we could handle all mesh types. Also the code style (eg braces) was not matching ours completely. Rather than complain at you I made a pass over the code to address these issues and added an ar weighted variable to the bayesian_ar_detect_app_test. I'll push these now

**Comment by burlen** [comment]
If you're curious to see what I changed you could in the "files changed tab" view "changes from" menu in the top left to select the individual commits I made.

**Comment by taobrienlbl** [comment]
Thank you @burlen for manually fixing these issues!  Glad to see it otherwise seemed reasonable.  I do need to ask you at some point about any command(s) you use to check style.

**Comment by burlen** [comment]
oops, my changes broke your new test, will push fix now

**Pull request merged**